# Exercise 6.01: Using Basic Operations and TensorFlow Constants

In this exercise, we will be using arithmetic operations in TensorFlow to emulate an artificial neuron by performing a matrix multiplication and addition, and applying a non-linear function, sigmoid.

The following steps will help you complete the exercise:

1.- Import the `tensorflow` package as `tf`

In [1]:
import tensorflow as tf

2.- Create a tensor called `W` of shape $[1,6]$ (that is, with 1 row and 6 columns), using `tf.constant()`, that contains the matrix $[1.0, 2.0, 3.0, 4.0, 5.0, 6.0]$. Print its value:

Output:

```
<tf.Tensor: shape=(1, 6), dtype=float32, numpy=array([[1., 2., 3., 4., 5., 6.]], dtype=float32)>
```

In [2]:
W = tf.constant([1.0, 2.0, 3.0, 4.0, 5.0, 6.0], shape=[1, 6]) 
W

<tf.Tensor: shape=(1, 6), dtype=float32, numpy=array([[1., 2., 3., 4., 5., 6.]], dtype=float32)>

3.- Create a tensor called X of shape $[6,1]$ (that is, with 6 rows and 1 column), using `tf.constant()`, that contains $[7.0, 8.0, 9.0, 10.0, 11.0, 12.0]$. Print its value

Output:

```
<tf.Tensor: shape=(6, 1), dtype=float32, numpy=
array([[ 7.],
       [ 8.],
       [ 9.],
       [10.],
       [11.],
       [12.]], dtype=float32)>
```

In [3]:
X = tf.constant([7.0, 8.0, 9.0, 10.0, 11.0, 12.0], shape=[6, 1]) 
X

<tf.Tensor: shape=(6, 1), dtype=float32, numpy=
array([[ 7.],
       [ 8.],
       [ 9.],
       [10.],
       [11.],
       [12.]], dtype=float32)>

4.- Create a tensor called `b`, using `tf.constant()`, that contains $-88$. Print its value:

```
<tf.Tensor: shape=(), dtype=float32, numpy=-88.0>
```

In [4]:
b = tf.constant(-88.0)
b

<tf.Tensor: shape=(), dtype=float32, numpy=-88.0>

5.- Perform a matrix multiplication between `W` and `X` using `tf.matmul`, save its results in the `mult` variable, and print its value

Output

```
<tf.Tensor: shape=(1, 1), dtype=float32, numpy=array([[217.]], dtype=float32)>
```

In [5]:
mult = tf.matmul(W, X)
mult

<tf.Tensor: shape=(1, 1), dtype=float32, numpy=array([[217.]], dtype=float32)>

6.- Perform a matrix addition between `mult` and `b`, save its results in a variable called `Z`, and print its value:

Output:

```
<tf.Tensor: shape=(1, 1), dtype=float32, numpy=array([[305.]], dtype=float32)>
```

In [6]:
Z = mult + b
Z

<tf.Tensor: shape=(1, 1), dtype=float32, numpy=array([[129.]], dtype=float32)>

7.- Apply the `sigmoid` function to `Z` using `tf.math.sigmoid`, save its results in a variable called a, and print its value. The `sigmoid` function transforms any numerical value within the range $0$ to $1$ (we will learn more about this in the following sections):

Output:

```
<tf.Tensor: shape=(1, 1), dtype=float32, numpy=array([[1.]], dtype=float32)>
```

In [7]:
a = tf.math.sigmoid(Z)
a

<tf.Tensor: shape=(1, 1), dtype=float32, numpy=array([[1.]], dtype=float32)>

The `sigmoid` function has transformed the original value of `Z`, which was 129, to 1.